planetscale/passwords.go

```diff
@@ -0,0 +1,159 @@
+package planetscale
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+type ConnectionStrings struct {
+	DotNet   string `json:"dotnet"`
+	General  string `json:"general"`
+	MySQLCLI string `json:"mysqlcli"`
+	PHP      string `json:"php"`
+	Prisma   string `json:"prisma"`
+	Rails    string `json:"rails"`
+	GoLang   string `json:"go"`
+	Java     string `json:"java"`
+	Rust     string `json:"rust"`
+}
+
+type DatabaseBranchPassword struct {
+	PublicID          string            `json:"id"`
+	Name              string            `json:"display_name"`
+	UserName          string            `json:"username"`
+	Role              string            `json:"role"`
+	Branch            DatabaseBranch    `json:"database_branch"`
+	CreatedAt         time.Time         `json:"created_at"`
+	DeletedAt         time.Time         `json:"deleted_at"`
+	PlainText         string            `json:"plain_text"`
+	ConnectionStrings ConnectionStrings `json:"connection_strings"`
+}
+
+// DatabaseBranchPasswordRequest encapsulates the request for creating/getting/deleting a
+// database branch password.
+type DatabaseBranchPasswordRequest struct {
+	Organization string `json:"-"`
+	Database     string `json:"-"`
+	Branch       string `json:"-"`
+	DisplayName  string `json:"display_name"`
+}
+
+// ListDatabaseBranchPasswordRequest encapsulates the request for listing all passwords
+// for a given database branch.
+type ListDatabaseBranchPasswordRequest struct {
+	Organization string
+	Database     string
+	Branch       string
+}
+
+// GetDatabaseBranchPasswordRequest encapsulates the request for listing all passwords
+// for a given database branch.
+type GetDatabaseBranchPasswordRequest struct {
+	DatabaseBranchPasswordRequest
+	PasswordId string
+}
+
+// DeleteDatabaseBranchPasswordRequest encapsulates the request for deleting a password
+// for a given database branch.
+type DeleteDatabaseBranchPasswordRequest struct {
+	DatabaseBranchPasswordRequest
+	PasswordId string
+}
+
+// DatabaseBranchPasswordsService is an interface for communicating with the PlanetScale
+// Database Branch Passwords API endpoint.
+type DatabaseBranchPasswordsService interface {
```

Comment:
I think we call it `PasswordsService` as it's simpler. What do you think? I don't have a strong opinion on this though.